lib/controllers/share.js

```diff
@@ -23,22 +25,37 @@ function makeGist(serviceOptions, body) {
     if (serviceOptions.accessToken !== undefined) {
         headers['Authorization'] = 'token ' + serviceOptions.accessToken;
     }
-    return requestp({
+    return got({
         url: gistAPI,
         method: 'POST',
         headers: headers,
-        json: true,
-        body: {
+        json: {
             files: gistFile,
             description: (serviceOptions.gistDescription || 'User-created catalog'),
             public: false
-        }, transform: function(body, response) {
-            if (response.statusCode === 201) {
-                console.log('Created ID ' + response.body.id + ' using Gist service');
-                return response.body.id;
-            } else {
-                return response;
-            }
+        },
+        hooks: {
+            afterResponse: [
+                (response, _retryWithMergedOptions) => {
+                    if (response.statusCode === 201) {
+                        console.log('Created ID ' + response.body.id + ' using Gist service');
+                        return response.body.id;
```

Comment:
Doubt this is tested by the test suite too, and I have no idea if body has an id attribute with got.